kong/plugins/prometheus/exporter.lua

```diff
@@ -190,10 +207,26 @@ if kong_subsystem == "http" then
       route_name = message.route.name or message.route.id
     end
 
+    local consumer = ""
+    if message and serialized.consumer ~= nil then
+      consumer = serialized.consumer
+    end
+
     labels_table[1] = service_name
     labels_table[2] = route_name
     labels_table[3] = message.response.status
-    metrics.status:inc(1, labels_table)
+    labels_table[4] = consumer
+
+    labels_table_status[1] = service_name
+    labels_table_status[2] = route_name
+    labels_table_status[3] = message.response.status
+    labels_table_status[4] = kong.response.get_source()
```

Comment:
Please read the documentation of the function: https://docs.konghq.com/gateway/latest/pdk/kong.response/#kongresponseget_source.
Please parse the string and then populate the source accordingly.

Reply:
The end-user of the metric will have no clue what is the difference between "exit" or "error". Avoid giving out details about the product as much as possible - the more this happens the more is cognitive load on the end user. Keep it simple.

Reply:
@hbagdi @sichvoge there are 3 possible values for `kong.response.get_source()`: `exit` (when the response originates in kong, e.g. due to invalid credentials), `error` (when the response originates in kong due to an error, e.g. unreachable upstream) and `service` (when response originates in the upstream).
given the above, are the translations below adequate?
`exit` -> `kong`
`error` -> `kong_due_to_error`
`service` -> `service`

Reply:
exit -> kong
error -> kong
service -> service

Let's not doll out more information than necessary at the moment.

Reply:
We can provide more fine-grained data to the user in the future if there is a need. When in doubt, start with the narrowest possible API.